### PR TITLE
TOURCMS-10033 add feature payments for bookings

### PR DESCRIPTION
**Pull request by palisis-juancarlos**

### CODE REVIEW

File: controllers/ReservationSystem.php [hunk 1]
```diff
@@ -295,31 +295,12 @@ public function showBooking($channelId, $bookingId)
     {
         $this->listChannels();
```
Comment (palisis-juanra) [inline controllers/ReservationSystem.php]: same than other times. Has no sense call list channels

File: controllers/ReservationSystem.php [hunk 2]
```diff
@@ -328,14 +309,47 @@ public function cancelBooking($channelId, $bookingId)
...
 
         // Call TourCMS API, cancelling the booking
         $result = $tourCMSOperator->cancel_booking($booking, $channelId);
 
         return $result->booking;
     }
 
+    public function showCustomer($customerId, $channelId)
+    {
+        $this->listChannels();
```
Comment (palisis-juanra) [inline controllers/ReservationSystem.php]: same

File: controllers/ReservationSystem.php [hunk 3]
```diff
@@ -328,14 +309,47 @@ public function cancelBooking($channelId, $bookingId)
...
+        $operator = $this->createOperator($channelId);
+        $customerDetails = $operator->show_customer($customerId, $channelId);
+        return $customerDetails;
+
+    }
+
+    public function updateCustomer($channelId, $customerData)
+    {
+        $this->listChannels();
```
Comment (palisis-juanra) [inline controllers/ReservationSystem.php]: same

File: index.php [hunk 4]
```diff
@@ -48,6 +52,7 @@
         $data['content']['bookingChannel'] = $booking->channel_id;
         $data['content']['bookingId'] = $booking->booking_id;
     } elseif (isset($_POST['postRequesBooking'])) {
+        $reserSys->listChannels();
```
Comment (palisis-juanra) [inline index.php]: as said before, has no sense to be listChannels

**Review by palisis-juanra — CHANGES_REQUESTED** [review]

**Review by palisis-juanra — APPROVED** [review]

File: views/partials/inputSelectCountries.html [hunk 5]
```diff
@@ -1,4 +1,4 @@
-<option value="" selected disabled>Select a country</option>
+<option value=""selected disabled>Select a country</option>
```
Comment (palisis-juanra) [inline views/partials/inputSelectCountries.html]:
```suggestion
<option value="" selected disabled>Select a country</option>
```

Comment (palisis-juanra) [reply views/partials/inputSelectCountries.html]: please, do not remove the space between the field

File: views/updateCustomer.html [hunk 6]
```diff
@@ -20,7 +20,7 @@
                     <div class="form-group col mx-2 mt-3">
                         <label for="firstname">First Name</label>
                         <input type="text" class="form-control" id="firstname" name="postInfoCustomer[firstname]"
-                            required value="{{firstname}}">
+                             value="{{firstname}}">
```
Comment (palisis-juanra) [inline views/updateCustomer.html]: be careful with new lines

**Review by palisis-juanra — CHANGES_REQUESTED** [review]

File: views/updateCustomer.html (hunk 6 above)
Comment (palisis-juancarlos) [reply views/updateCustomer.html]: fixed

File: views/partials/inputSelectCountries.html (hunk 5 above)
Comment (palisis-juancarlos) [reply views/partials/inputSelectCountries.html]: fixed

File: views/updateCustomer.html [hunk 7]
```diff
@@ -20,7 +20,7 @@
                     <div class="form-group col mx-2 mt-3">
                         <label for="firstname">First Name</label>
                         <input type="text" class="form-control" id="firstname" name="postInfoCustomer[firstname]"
-                            required value="{{firstname}}">
+                            value="{{firstname}}">
```
Comment (palisis-juanra) [inline views/updateCustomer.html]: this should be in the same input line

**Review by palisis-juanra — CHANGES_REQUESTED** [review]

File: index.php [hunk 8]
```diff
@@ -58,19 +58,18 @@
...
-            $data['content']['bookingDetailsError'] = $e->getMessage();
+    if (isset($_POST['postSearchBooking'])) {
+        if (isset($_POST['postUpdateCancelation'])) {
+            $cancelation = $reserSys->cancelBooking($_POST['postChannelId'], $_POST['postBookingId'], $_POST['postCancelationReason']);
         }
+        if(isset($_POST['postCreatePayment'])){
+            $payment = $reserSys->createPayment($_POST['postChannelId'], $_POST['postInfoPayment']);
+            }
+        $data['content']['bookingDetails'] = $reserSys->showBooking($_POST['postChannelId'], $_POST['postBookingId']);
+        $data['content']['bookingDetails']['auxBooleanPaymentFulfilled'] =  $data['content']['bookingDetails']['payment_status'] == 3 ? 1 : 0;
```
Comment (palisis-juanra) [inline index.php]: what is 3?
Please use user friendly const instead of magic number / words

**Review by palisis-juanra — CHANGES_REQUESTED** [review]

File: index.php (hunk 8 above)
Comment (palisis-juancarlos) [reply index.php]: fixed

**Review by palisis-juanra — APPROVED** [review]

**Pull request merged**